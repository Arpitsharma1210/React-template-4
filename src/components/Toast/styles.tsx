import { styled } from "styled-components";
import { fontSize, fontWeight } from "../../theme/style.typography";
import { colors } from "../../theme/style.palette";

export const StyledToastContainer = styled.div`
  display: flex;
  align-items: center;
  margin: 0;

  figure {
    width: 32px;
    height: 32px;
    margin: 0;
    margin-right: 12px;

    img {
      display: block;
      width: 100%;
      height: 100%;
    }
  }

  .toast-text {
    margin-right: 75px;

    &-maintext {
      font-size: ${fontSize.b1};
      font-weight: ${fontWeight.medium};
      color: ${colors.black15};
      margin: 0;
    }

    &-subtext {
      font-size: ${fontSize.b2};
      color: ${colors.black20};
      margin: 0;
    }
  }

  .closeBtn {
    align-self: flex-start;
  }
`;
