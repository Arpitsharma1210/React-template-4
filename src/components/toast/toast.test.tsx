import React from 'react';
import { render, screen } from '@testing-library/react';
import Toast from './'; 
import '@testing-library/jest-dom';

describe('Toast Component', () => {

  it('renders text in ToastInfoText when text prop is provided', () => {
    render(<Toast text="Success Message" />);
    expect(screen.getByText("Success Message")).toBeInTheDocument();
  });

  it('renders subText in ToastInfoSubText when subText prop is provided', () => {
    render(<Toast subText="Additional Information" />);
    expect(screen.getByText("Additional Information")).toBeInTheDocument();
  });

  it('renders both text and subText when both props are provided', () => {
    render(<Toast text="Success Message" subText="Additional Information" />);
    expect(screen.getByText("Success Message")).toBeInTheDocument();
    expect(screen.getByText("Additional Information")).toBeInTheDocument();
  });

  it('matches snapshot when given props', () => {
    const { asFragment } = render(
      <Toast
        text="Success Message"
        subText="Additional Information"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
