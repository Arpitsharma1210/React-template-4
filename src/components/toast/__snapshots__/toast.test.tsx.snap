// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast Component matches snapshot when given props 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnyFh jeWNUc"
  >
    <figure>
      <img
        alt="toast icon"
        src="test-file-stub"
      />
    </figure>
    <div
      class="toast-text"
    >
      <p
        class="toast-text-maintext"
      >
        Success Message
      </p>
      <p
        class="toast-text-subtext"
      >
        Additional Information
      </p>
    </div>
    <div
      class="closeBtn"
      role="button"
      tabindex="0"
    >
      <img
        alt="close icon"
        src="test-file-stub"
      />
    </div>
  </div>
</DocumentFragment>
`;
